src/firmware/boards/legacy_robot_stm32f4/io/receive.c

```diff
@@ -105,13 +108,19 @@ static void receive_task(void *UNUSED(param))
                 {
                     // Broadcast frame must contain a camera packet or drive packet
                     // Note that camera packets have a variable length.
-                    if (dma_buffer[MESSAGE_PURPOSE_ADDR] == 0x0FU)
+                    if (dma_buffer[MESSAGE_PURPOSE_INDEX] == 0x0FU)
```

Comment:
I assume that 0x0FU and 0x10U below correspond to either camera or drive packets. Might be good to explicitly clarify that in brackets in the comment.

Reply:
in the interests of limiting scope I'm not going to start adding comments here, but I think that given we're only calling one function in each `if` branch it's fairly obvious?